### ISSUE TITLE
Subscription Identifier Bug

### ISSUE THREAD

**Issue by stefanodaroit** [vernemq/vernemq#1992]
### Environment

- VerneMQ Version: vernemq/vernemq:1.12.3 docker image
- OS: build=22000  caption=Microsoft Windows 11 Home  64-bit  version=10.0.22000

### Expected behaviour

As noted in the OASIS specification: “it is a Protocol Error for a PUBLISH packet to contain any Subscription Identifier other than those received in SUBSCRIBE packet which caused it to flow. A PUBLISH packet sent from a Client to a Server MUST NOT contain a Subscription Identifier [MQTT-3.3.4-6]”.

### Actual behaviour

If a client tries to send a PUBLISH packet that includes a “Subscription Identifier” the broker does not report a protocol error, it forwards the packet to subscribers including the property. This happens only if the subscribers do not set themselves a “Subscription Identifier“ for the subscription.

**Comment by ioolkos** [comment]
@stefanodaroit Yes, I guess we should add that & spend more CPU cycles to check the PUBLISH properties for spec conformity :)
Thanks!

---
:point_right: Thank you for supporting VerneMQ: https://github.com/sponsors/vernemq
:point_right: Using the binary VerneMQ packages commercially (.deb/.rpm/Docker) requires a paid subscription.
